### DISCUSSION TITLE
Can't get text from the pdf

### DISCUSSION THREAD

**Discussion by hemilparmar** [py-pdf/pypdf#1465] [Q&A] [answered]
when i parse this pdf i can not get the content of pdf
[test.pdf](https://github.com/py-pdf/PyPDF2/files/10133210/test.pdf)
[test2.pdf](https://github.com/py-pdf/PyPDF2/files/10133212/test2.pdf)

PyPDF2 version : 2.11.2

Code:

import PyPDF2 
pdfFileObj = open('test.pdf', 'rb') 
pdfReader = PyPDF2.PdfFileReader(pdfFileObj) 
pageObj = pdfReader.getPage(1) 
print(pageObj.extractText()) 
pdfFileObj.close()

**Comment by MartinThoma** [comment, marked as answer]
From https://pypdf2.readthedocs.io/en/latest/user/extract-text.html

> Sometimes PDFs do not contain the text as it’s displayed, but instead an image. You notice that when you cannot copy the text. Then there are PDF files that contain an image and a text layer in the background. That typically happens when a document was scanned. Although the scanning software (OCR) is pretty good today, it still fails once in a while. PyPDF2 is no OCR software; it will not be able to detect those failures. PyPDF2 will also never be able to extract text from images.

**Reply by hemilparmar** [reply]
Yes , can you see the test2.pdf? In that pdf file,  text is there but it gives the random numbers like some text encryption is there.

**Reply by pubpub-zz** [reply]
your pdf has been volontarily generated to prevent text extraction
have a look at #1171 for more details
